homeassistant/components/file_upload/__init__.py

```diff
@@ -0,0 +1,176 @@
+"""The File Upload integration."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Iterator
+from contextlib import contextmanager
+from dataclasses import dataclass
+from pathlib import Path
+import shutil
```

Comment:
Shouldn't we use an async variant to prevent blocking IO?

Reply:
This function is only called from sync methods executed in the executor. This function exists to do I/O (mutate filesystem) so there is no way around it.